### PR TITLE
refactor infectionswithfeedback.py to allow shared infection feedback strength across sites

### PR DESCRIPTION
**Pull request by sbidari**
Used `jnp.broadcast` instead of padding to get desired shape for `infection_feedback_strength`. This allows flexibility of having shared or individual infection feedback strength across different sites/sub populations which the previous version did not.

This change also makes `pad_edges_to_match` function from the `arrayutils` unnecessary but I left it for now.

### CODE REVIEW

**Review by dylanhmorris — APPROVED** [review]
This is a much better solution, thanks @sbidari!
